server/src/main/java/org/elasticsearch/script/field/BigIntegerField.java

```diff
@@ -9,9 +9,105 @@
 package org.elasticsearch.script.field;
 
 import java.math.BigInteger;
+import java.util.List;
+import java.util.stream.Collectors;
 
 public class BigIntegerField extends Field<java.math.BigInteger> {
 
+    /* ---- Conversion Class From Other Fields ----*/
+
+    /**
+     * Convert to a {@link BigIntegerField} from Long, Double or String Fields.
+     * Longs and Doubles are wrapped as BigIntegers.
+     * Strings are parsed as either Longs or Doubles and wrapped in a BigInteger.
+     */
+    public static final Converter<BigInteger, BigIntegerField> BigInteger;
+
+    static {
+        BigInteger = new Converter<BigInteger, BigIntegerField>() {
+            @Override
+            public BigIntegerField convert(Field<?> sourceField) {
+                if (sourceField instanceof LongField) {
+                    return LongField.convertLongToBigIntegerField((LongField) sourceField);
+                }
+
+                if (sourceField instanceof DoubleField) {
+                    return DoubleField.convertDoubleToBigIntegerField((DoubleField) sourceField);
+                }
+
+                if (sourceField instanceof StringField) {
+                    return StringField.convertStringToBigIntegerField((StringField) sourceField);
+                }
+
+                if (sourceField instanceof DateMillisField) {
+                    return LongField.convertLongToBigIntegerField(
+                            DateMillisField.convertDateMillisToLongField((DateMillisField) sourceField));
+                }
+
+                if (sourceField instanceof DateNanosField) {
+                    return LongField.convertLongToBigIntegerField(
```

Comment:
What's the thinking here?

Reply:
I dropped this a line to fit into our character limits, but functionally didn't change anything other than the location of the method.

Reply:
Moved the line back up with the name changes, but did not change the logic.